### ISSUE TITLE
github dependency comes incomplete

### ISSUE THREAD

**Issue by DanielKucal** [npm/npm#14420]
#### I'm opening this issue because:
- [ ] npm is crashing.
- [x] npm is producing an incorrect install.
- [x] npm is doing something I don't understand.
- [ ] Other (_see below for feature requests_):
#### What's going wrong?

Hi, I have npm github dependency defined like this:

```
{
    "dependencies": {
        "ng2-file-upload": "git+https://github.com/valor-software/ng2-file-upload#development"
    }
}
```

And it downloads but doesn't include any files in subdirectories, with the only one exception which is `components/file-upload/readme.md`. The same thing works with direct npm dependency, but I actually need to use a fork, which doesn't work too... 
What I've already tried:
- branches development and master
- tag v1.1.3-0
- `--no-bin-links`
- deleting module
- http, https and ssh
- `npm install`, `npm update`
- updating npm
- `prepublish` script without the `npm run flow.clean` command
- a hammer

But none of the above gives any result nor console warnings are printed.

Here is the repository I'm trying to include into my project: https://github.com/valor-software/ng2-file-upload
and its current package.json: https://gist.github.com/DanielKucal/0e2a09534c03e9a191d676cfb9e8fe9d
#### How can the CLI team reproduce the problem?

`npm install https://github.com/valor-software/ng2-file-upload#development --save`
### supporting information:
- `npm -v` prints: 3.10.9
- `node -v` prints: v5.11.1
- `npm config get registry` prints: https://registry.npmjs.org/
- Windows, OS X/macOS, or Linux?: Windows 8.1
- Network issues:
  - Geographic location where npm was run: Middle Europe
  - [ ] I use a proxy to connect to the npm registry.
  - [ ] I use a proxy to connect to the web.
  - [ ] I use a proxy when downloading Git repos.
  - [ ] I access the npm registry via a VPN
  - [ ] I don't use a proxy, but have limited or unreliable internet access.
- Container:
  - [ ] I develop using Vagrant on Windows.
  - [ ] I develop using Vagrant on OS X or Linux.
  - [ ] I develop / deploy using Docker.
  - [ ] I deploy to a PaaS (Triton, Heroku).

The question on stackoverflow: http://stackoverflow.com/questions/40251756/npm-github-dependency-came-without-files-in-subdirectories
Thanks in advance!

**Comment by othiym23** [comment]
This package isn't meant to be installed via git dependency -- if you look at the `.gitignore` and `.npmignore` files, when you run the npm install via Git dependency, they're going to end up ignoring all of each other's useful source files. If you're hacking on this package, then you're going to want to clone it, and then `npm link` it into wherever you're using it until the new version has been published to a registry.

**Comment by DanielKucal** [comment]
Thanks for response. I've already looked at these ignore files and removed source files from there. It actually made it work, but the cloned module had to be compiled everytime I used `npm install` or update.  This solution doubled compilation time of my project. `npm link` works only locally.
What can I do if I need a bit modified Typescript repository? Is there any other way to go except the time-wasting solution from above or keeping compiled files in a Github repository?
